crypto/x509/x509_vfy.c

```diff
@@ -1754,119 +1754,70 @@ int X509_cmp_current_time(const ASN1_TIME *ctm)
 
 int X509_cmp_time(const ASN1_TIME *ctm, time_t *cmp_time)
 {
-    char *str;
+    static const size_t utctime_length = sizeof("YYMMDDHHMMSSZ") - 1;
+    static const size_t generalizedtime_length = sizeof("YYYYMMDDHHMMSSZ") - 1;
+
     ASN1_TIME atm;
-    long offset;
-    char buff1[24], buff2[24], *p;
-    int i, j, remaining;
+    unsigned char time1[24], time2[24];
```

Comment:
This has gone.

I decided to bite the bullet and rewrite this with proper API functions. The reason I didn't do this initially is that I didn't want to bring any additional exposure of the old ASN.1 code into certificate handling. But, once we verify that the format is fixed and valid, it's _probably_ ok...

If someone wants to attack `ASN1_TIME_diff` and clean it up as well, be my guest :)